src/instructlab/model/remove.py

```diff
@@ -0,0 +1,105 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+import shutil
+
+# Third Party
+import click
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.model.backends.backends import is_model_gguf, is_model_safetensors
+
+
+class RemovalError(Exception):
+    """Custom exception for removal errors."""
+    pass
+
+
+def remove_item(model_path: Path, model_name: str, item_type: str):
```

Comment:
Why do you need `item_type` can't we infer the type somehow? Also 'type' is a bit misleading since someone could think about a language type (str, int etc)